src/auth/Passwordless.ts

```diff
@@ -125,14 +172,94 @@ export default class Passwordless extends BaseAuthAPI {
         headers: {
           'Content-Type': 'application/json',
         },
-        body: (await this.addClientAuthentication(
+        body: await this.addClientAuthentication(
           { client_id: this.clientId, connection: 'sms', ...bodyParameters },
           false
-        )) as Record<string, string>,
+        ),
       },
       initOverrides
     );
 
     return VoidApiResponse.fromResponse(response);
   }
+
+  /**
+   * Once you have a verification code, use this endpoint to login the user with their email and verification code.
+   *
+   * @example
+   * ```js
+   * const auth0 = new AuthenticationApi({
+   *    domain: 'my-domain.auth0.com',
+   *    clientId: 'myClientId',
+   *    clientSecret: 'myClientSecret'
+   * });
+   *
+   * await auth0.passwordless.loginWithEmailCode({
+   *   email: 'foo@example.com',
+   *   code: 'ABC123'
+   * });
+   * ```
+   */
+  async loginWithEmailCode(
+    bodyParameters: LoginWithEmailCodeRequest,
+    initOverrides?: InitOverride
+  ): Promise<VoidApiResponse> {
+    validateRequiredRequestParams(bodyParameters, ['email', 'code']);
+
+    const { email: username, code: otp, ...otherParams } = bodyParameters;
```

Comment:
Just curious, not saying we have to. But do you think we should we stick to the actual API names in bodyParams and use `username` and `otp` instead of `email` and `code` ? Will help us be closer to the actual API docs.

Reply:
Yeah, I know what you mean. I did want to do that. But that would mean you'd pass your phone number as `username` in `loginWithSMSCode`.

I think we could do this, but we should then flatten `loginWithEmailCode` and `loginWithSMSCode` to just `loginWithCode` and accept `username` which would be `The user's phone number if realm=sms, or the user's email if realm=email.` per the docs. wdyt?

Reply:
Honestly, I like the idea of not inventing our own API on top of the actual Auth0 API when not needed. When it improves DX enought to warrant that, I'd do that. But here, I'd lean towards going with `loginWithCode`, or should it be `loginWithOtp` 🤔  ?

Reply:
Went with `loginWithEmail`, `loginWithSMS`